### PR TITLE
change array.length issue in filters

### LINKED ISSUE #9855: Code quality refactoring
I was doing a code quality analysis with [SonarQube](www.sonarqube.org) on Bokeh and came through several refactoring candidates that may be addressed quickly and I believe would benefit the overall project. Some of them may be minor bugs but still.
Here are the main issues (that may be repeated througout the code)

### Readability issues

- Nested ```if``` statements that may be merged:

```python
if condition1:
    if condition2:
        # ...
```
to 
```python
if condition1 and condition2:
        # ...
```

### Maintainability Issues

- Repeated constants should be assigned to variable for easier future refactoring. 

### Efficiency

- In cycles (for/while) the use of if/elif and continue/break to improve performance and excecution time when its not necesary to excecute the remaining code

### Cognitive complexity

- Some functions are very cognitively complex, some of them could be eithermodularized or simplified. Some of the issues mentioned beforehand even reduce complexity in some of them.

### Repeated turnouts in conditions

[plots.py](https://github.com/bokeh/bokeh/blob/master/bokeh/models/plots.py#L357) has the same result for the if and elif conditions. It can be replaced by the ```or``` operator as its more efficient and clear. (or stops checking the remaining conditions once it finds a ```True```) run this code if you dont believe me:
```python
def f():
	print("text")
	return False
print(True or f())
```

### switch case issues

- Some switch functions are missing the break after each ```case``` test, making the switch statement lose its puropose. (both cases are completely different) 
- ~Also, when evaluating ```case a || b```, only ```a``` is considered. Should be wrapped around parenthesis or considered as two separate cases [templating.js line 35](https://github.com/bokeh/bokeh/blob/master/bokehjs/src/lib/core/util/templating.ts#L35)~

### indent and curly braces in if/else {}

This is just a good practice and a very minor issue, but it improves readability for future maintainers

### Missing type in [```palettes.ts```](https://github.com/bokeh/bokeh/blob/master/bokehjs/src/lib/api/palettes.ts)
- Set3 considers Set3_11 twice and doesnt consider Set3_12

####  length >= 0
- In [index_filters.ts](https://github.com/bokeh/bokeh/blob/master/bokehjs/src/lib/models/filters/index_filter.ts) line 32 an array length is checked to be greater **or equal** than 0... Is this a mistake or on purpose?

Is it alright if I fix these issues and submit a PR? If any of them are intentional for some reason, please let me know :+1:

### PR DESCRIPTION
**Pull request by auirarrazaval**
addresses #9855

### CODE REVIEW

File: bokehjs/src/lib/models/filters/filter.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ export class Filter extends Model {
 
   compute_indices(_source: DataSource): number[] | null {
     const filter = this.filter
-    if (filter != null && filter.length >= 0) {
+    if (filter != null && filter.length > 0) {
```
Comment (mattpap) [inline bokehjs/src/lib/models/filters/filter.ts:33]: The other clause should have been removed, otherwise users will get a misleading warning.

Comment (auirarrazaval) [reply bokehjs/src/lib/models/filters/filter.ts:33]: Do you mean ```filter != null```?

Comment (mattpap) [reply bokehjs/src/lib/models/filters/filter.ts:33]: No, I mean the other clause. The final code should be:
```ts
if (filter != null) {
```

Comment (auirarrazaval) [reply bokehjs/src/lib/models/filters/filter.ts]: @mattpap I've deleted the clause

File: bokehjs/src/lib/models/filters/index_filter.ts [hunk 2]
```diff
@@ -29,7 +29,7 @@ export class IndexFilter extends Filter {
   }
 
   compute_indices(_source: DataSource): number[] | null {
-    if (this.indices != null && this.indices.length >= 0) {
+    if (this.indices != null && this.indices.length > 0) {
```
Comment (bryevdv) [inline bokehjs/src/lib/models/filters/index_filter.ts:32]: So this change to a strict comparision breaks things for zero length arrays (which is a corner case, but should function). I think the second condition should just be removed altogether.

Comment (auirarrazaval) [reply bokehjs/src/lib/models/filters/index_filter.ts:32]: @bryevdv I did on my latest commit, but the js-test check is still failing. Locally I ran all the tests and they passed.

**Pull request merged**